packages/kokkos-kernels/src/sparse/impl/KokkosSparse_twostage_gauss_seidel_impl.hpp

```diff
@@ -0,0 +1,907 @@
+/*
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 3.0
+//       Copyright (2020) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Siva Rajamanickam (srajama@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#ifndef _KOKKOS_TWOSTAGE_GS_IMP_HPP
+#define _KOKKOS_TWOSTAGE_GS_IMP_HPP
+
+#include <KokkosKernels_config.h>
+#include "Kokkos_Core.hpp"
+
+// Blas Kernels
+#include "KokkosBlas1_scal.hpp"
+#include "KokkosBlas1_mult.hpp"
+#include "KokkosBlas1_axpby.hpp"
+#include "KokkosSparse_spmv.hpp"
+#include "KokkosKernels_SimpleUtils.hpp"
+
+// needed for classical GS
+#include "KokkosSparse_sptrsv.hpp"
+#include "KokkosKernels_SparseUtils.hpp"
+
+#include "KokkosSparse_gauss_seidel_handle.hpp"
+
+#define KOKKOSSPARSE_IMPL_TWOSTAGE_GS_MERGE_SPMV
```

Comment:
Can this macro be replaced with a runtime option if it needs to be configurable, or just removed if it doesn't? This would be for kokkos-kernels, not necessarily for this PR.